envisage/examples/demo/plugins/tasks/attractors/plot_2d_pane.py

```diff
@@ -34,29 +36,63 @@ class Plot2dPane(TraitsTaskPane):
     y_data = Property(observe="active_model.y_data")
     x_label = Property(Str, observe="active_model.x_label")
     y_label = Property(Str, observe="active_model.y_label")
+    plot = Instance(Plot)
+
+    def _plot_default(self):
+        plot = Plot(ArrayPlotData(x=self.x_data, y=self.y_data))
+        plot.x_axis.title = self.x_label
+        plot.y_axis.title = self.y_label
+
+        plot.plot(
+            ("x", "y"),
+            type=self.plot_type,
+            name=self.title,
+            marker='pixel',
+            color="blue"
+        )
+
+        return plot
+
+    @observe("x_data,y_data")
+    def _update_plot_data(self, event):
+        if event.name == "x_data":
+            self.plot.data.set_data("x", event.new)
+        else:
+            self.plot.data.set_data("y", event.new)
+        self.plot.invalidate_and_redraw()
+
+    @observe("x_label,y_label")
+    def _update_axis_label(self, event):
+        if event.name == "x_label":
+            self.plot.x_axis.title = event.new
+        else:
+            self.plot.y_axis.title = event.new
+        self.plot.invalidate_and_redraw()
+
+    @observe("active_model")
+    def _update_plot_new_model(self, event):
+        if event.old:
+            self.plot.delplot(event.old.name)
```

Comment:
The old active_model's name is the old `title` that was used as the name for the previous plot.

I do think this naming is a little confusing, as `title` is not actually used as a title... In fact, prior to this PR it was not actually used at all!  I found it convenient for naming plots so they could be easily deleted when we switch models.

Note that if we do not delete the old plot when we change model, things appear to work perfectly fine initially, however if you change models a few times eventually you see the
```
OverflowError: Exceeded cell block limit
Abort trap: 6
``` 
crash as mentioned in another comment.  Deleting old plots prevents this